test/common/common/thread_test.cc

```diff
@@ -190,6 +203,41 @@ TEST_F(ThreadAsyncPtrTest, ManagedAlloc) {
   }
 }
 
+TEST_F(ThreadAsyncPtrTest, TruncateWait) {
+  absl::Notification notify;
+  auto thread = thread_factory_.createThread([&notify]() { notify.WaitForNotification(); },
+                                             Options{"this name is way too long for posix"});
+  notify.Notify();
+
+  // To make this test work on multiple platforms, just assume the first 10 characters
+  // are retained.
+  EXPECT_THAT(thread->name(), testing::StartsWith("this name "));
+  thread->join();
+}
+
+TEST_F(ThreadAsyncPtrTest, TruncateNoWait) {
+  auto thread =
+      thread_factory_.createThread([]() {}, Options{"this name is way too long for posix"});
+
+  // To make this test work on multiple platforms, just assume the first 10 characters
+  // are retained.
+  EXPECT_THAT(thread->name(), testing::StartsWith("this name "));
+  thread->join();
+}
+
+TEST_F(ThreadAsyncPtrTest, NameNotSpecifiedWait) {
```

Comment:
I feel like this test is not needed.

Reply:
this is the only test that hits a code-path where we didn't specify a name, but instead read it from the OS and store it in the Thread object.

Reply:
Oh, I see what code that's testing. Ok, good to have it then.